src/Installer/redist-installer/targets/GenerateLayout.targets

```diff
@@ -202,6 +202,17 @@
         <RelativeLayoutPath>packs/%(PackageName)/%(PackageVersion)</RelativeLayoutPath>
       </BundledLayoutPackage>
 
+      <BundledLayoutPackage Include="MicrosoftDotNetILCompilerPackNupkg" Condition="'$(BundleNativeAotCompiler)' == 'true'">
+        <PackageName>runtime.$(SharedFrameworkRid).Microsoft.DotNet.ILCompiler</PackageName>
+        <PackageVersion>$(MicrosoftNETCoreAppRuntimePackageVersion)</PackageVersion>
+        <TargetFramework>$(TargetFramework)</TargetFramework>
+        <RelativeLayoutPath>packs/%(PackageName)/%(PackageVersion)</RelativeLayoutPath>
+      </BundledLayoutPackage>
+
+      <BundledLayoutLibraryPackage Include="$(SourceBuiltShippingPackagesDir)/../runtime/Microsoft.DotNet.ILCompiler.$(MicrosoftNETCoreAppRuntimePackageVersion).nupkg" Condition="'$(BundleNativeAotCompiler)' == 'true'" />
```

Comment:
The `SourceBuiltShippingPackagesDir` property isn't available outside of the VMR.

Reply:
> isn't available outside of the VMR

I think for non-VMR builds we can set `BundleNativeAotCompiler=false`?

Reply:
We want to minimize the difference between a repo source-build inside the VMR and inside the repository as otherwise things easily break. I understand that we want to get this in before the next code complete date so I'm fine with disabling this for now in the repo build but please file a tracking issue. I don't think anything hinders repo source-build from executing this logic. It just needs a different path property.

Reply:
`DotNetBuildOrchestrator` is the property to condition on. https://github.com/dotnet/arcade/blob/main/Documentation/UnifiedBuild/Unified-Build-Controls.md#context-controls

Reply:
> We want to minimize the difference between a repo source-build inside the VMR and inside the repository as otherwise things easily break. I understand that we want to get this in before the next code complete date so I'm fine with disabling this for now in the repo build but please file a tracking issue. I don't think anything hinders repo source-build from executing this logic. It just needs a different path property.

I had asked that question here: https://github.com/dotnet/sdk/pull/41198#discussion_r1616868960. I don't know what the alternative looks like. I'll create an issue for it.

> DotNetBuildOrchestrator is the property to condition on. https://github.com/dotnet/arcade/blob/main/Documentation/UnifiedBuild/Unified-Build-Controls.md#context-controls

ok, I'll include this for setting BundleNativeAotCompiler to true.